web-app/src/main/webapp/script/directions.js

```diff
@@ -0,0 +1,15 @@
+function getDirections(directionsService, directionsRenderer, origin, destination) {
```

Comment:
nit: why not just return your `result` variable and let the caller set the directions? It would match your function name more.